client/gcp/gcp.go

```diff
@@ -0,0 +1,212 @@
+package gcp
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/bytebase/dbcost/client"
+)
+
+// Client is the client struct
+type Client struct {
+	httpClient http.Client
+}
+
+// NewClient return a client
+func NewClient() Client {
+	return Client{
+		httpClient: http.Client{},
+	}
+}
+
+const rdsServiceID = "9662-B51E-5089"
+
+// this apiKey should be blocked before merging
+const apiKey = "AIzaSyAj36rmGvQ_3kBqd6Lp3nSzPDo2_nD-sdM"
+
+// PriceInfoEndpoint is the endpoint for all Cloud SQL services on GCP
+// For more information, please refer to https://cloud.google.com/billing/v1/how-tos/catalog-api
+var PriceInfoEndpoint = fmt.Sprintf("https://cloudbilling.googleapis.com/v1/services/%s/skus?key=%s", rdsServiceID, apiKey)
+
+type unitPrice struct {
+	CurrencyCode string  `json:"currencyCode"`
+	Unit         string  `json:"units"`
+	Nano         float64 `json:"nanos"`
+}
+
+type tieredRate struct {
+	UnitPrice unitPrice `json:"unitPrice"`
+}
+
+type pricingExpression struct {
+	UsageUnit      string        `json:"usageUnit"`
+	TieredRateList []*tieredRate `json:"tieredRates"`
+}
+
+type pricingInfo struct {
+	PricingExpression pricingExpression `json:"pricingExpression"`
+}
+
+type category struct {
+	ServiceDisplayName string `json:"serviceDisplayName"`
+	ResourceFamily     string `json:"resourceFamily"`
+	ResourceGroup      string `json:"resourceGroup"`
+	UsageType          string `json:"usageType"`
+}
+
+type offer struct {
+	ID                string        `json:"skuId"`
+	Description       string        `json:"description"`
+	Category          category      `json:"category"`
+	PricingInfo       []pricingInfo `json:"pricingInfo"`
+	ServiceRegionList []string      `json:"serviceRegions"`
+}
+
+type pricing struct {
+	OfferList     []*offer `json:"skus"`
+	NextPageToken string   `json:"nextPageToken"`
+}
+
+func httpHelper(nextpageToken string) ([]*offer, error) {
```

Comment:
httpHelper isn't a meaningful name. Since we're making a request to get offer, we can call it getOfferWithPageToken.

Also nextpageToken should be nextPageToken.